src_fun/main.cpp

```diff
@@ -3,46 +3,110 @@
 #include "../src_common/hash.h"
 #include "../src_common/integer.h"
 #include "api.h"
+#include "testing.h"
+
 
 EnterApplicationNamespace
 
 
+static const String TEST_DIRECTORY = "test"_s;
+
+
 extern "C" int main(int argc, char** argv)
 {
     /*add_log_handler([](String severity, String subsystem, String msg)
     {
         printf("%.*s\n", StringArgs(msg));
     });*/
 
-    Dynamic_Array<String> non_flag_args = {};
-    for (umm i = 1; i < argc; i++)
-        if (argv[i][0] != '-')
-            *reserve_item(&non_flag_args) = wrap_string(argv[i]);
+    String first_flag_arg = {};
+    if (argc > 1 && argv[1][0] == '-')
+        first_flag_arg = consume(wrap_string(argv[1]), 1);
+
+    if (first_flag_arg == "test_runner"_s)

```

Comment:
Merge -test_runner and -test into one flag, such that:

* `fun -test` runs all tests
* `fun -test math-int-div0 math-float-div0` only runs those two
* `fun -test test/basic.test.fun` runs all from that file
* `fun -test math-*` runs all matching the `math-*` wildcard

This is necessary to make this usable once there are a lot of tests. I only want to run the ones for compiler interop for example, not all the adds and subtracts and hash tables, ...

Reply:
`-test_runner` is just an entry point for the subprocess which runs the test. I will rename it to `-test_process` to make that more clear.

I agree with everything else.

Reply:
> `fun -test test/basic.test.fun` runs all from that file

Actually not doing this, you can run all tests from this file by just doing `fun -test basic`.